official/resnet/cifar10_main.py

```diff
@@ -228,7 +228,14 @@ def main(argv):
   flags = parser.parse_args(args=argv[1:])
 
   input_function = flags.use_synthetic_data and get_synth_input_fn() or input_fn
-  resnet_run_loop.resnet_main(flags, cifar10_model_fn, input_function)
+  classifier = resnet_run_loop.resnet_main(
+      flags, cifar10_model_fn, input_function)
+
+  # Export the model if desired
```

Comment:
I think it would be cleaner to handle this inside of resnet_main() and just pass in shape. That would dedupe some code and is more intuitive IMHO.

Reply:
Sigh. Yeah, I went back and forth on that, but in the end, I felt like weighing resnet_main down with another, totally separate task set that required an extra param was asking too much of it. Quick poll: what do the people in reviewing this PR think-- put more of the logic in resnet_main, or keep it separate?

Reply:
I'm ok with both ideas. The main function should contain all the functions IMO, and it would be nice to avoid duplicate code. If we keeping them separate, perhaps we should rename the function `resnet_main` to `train`, `train_and_evaluate`, `run_training` or similar.

Reply:
Two and a half people counts as quorum. Done.